robottelo/host_helpers/repository_mixins.py

```diff
@@ -0,0 +1,766 @@
+import inspect
+import sys
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from robottelo import constants
+from robottelo import manifests
+from robottelo.config import settings
+from robottelo.errors import DistroNotSupportedError
+from robottelo.errors import OnlyOneOSRepositoryAllowed
+from robottelo.errors import ReposContentSetupWasNotPerformed
+from robottelo.errors import RepositoryAlreadyCreated
+from robottelo.errors import RepositoryAlreadyDefinedError
+from robottelo.errors import RepositoryDataNotFound
+
+
+class RepositoryHelpers:
+    def __init__(self):
+        for name, obj in inspect.getmembers(sys.modules[__name__], inspect.isclass):
+            new_cls = type(name, (obj,), {'satellite': self._satellite})
+            setattr(self, name, new_cls)
```

Comment:
What if instead, this were a helper method that accepted a satellite object and loaded all of these classes into a dictionary it then returned? That would avoid the problem of forcing this as a parent class to add in the repository classes.

Maybe something like this?
```python
def get_repo_classes(sat_obj):
    return {
        name, type(name, (obj,), {'satellite': self._satellite})
        for name, obj in inspect.getmembers(sys.modules[__name__], inspect.isclass)
        if issubclass(obj, BaseRepository)
    }
```